src/Native/Runtime/EHHelpers.cpp

```diff
@@ -133,8 +133,12 @@ COOP_PINVOKE_HELPER(void, RhpCopyContextFromExInfo,
     pContext->Rip = pPalContext->IP;
     pContext->Rsp = pPalContext->Rsp;
     pContext->Rbp = pPalContext->Rbp;
+#ifndef UNIX_AMD64_ABI
     pContext->Rdi = pPalContext->Rdi;
     pContext->Rsi = pPalContext->Rsi;
+#else // !UNIX_AMD64_ABI
+    pContext->Rdx = pPalContext->Rdx;
```

Comment:
I might also encourage a quick refresher comment for the reader on the amd64 unix abi.  I.e. for windows, it would be something like 
`// param regs: rcx, rdx, r8, r9, scratch: rax (return val), preserved: rbp, rdi, rsi, rbx, r12-r15 `